aggregate.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+import sys
+import json
+import pprint
+from cHash import c_hash
+
+pp = pprint.PrettyPrinter(indent=4)
+HASH_COUNT = 6
+CPUS = 4
+
+
+def csv_tuple_to_dict(csv):
+    return dict((a, b) for a,b in [x.split('=') for x in csv.split(',')])
+
+fields = {'saddr', 'daddr', 'sport', 'dport', 'proto'}
+def five_tuples_are_equal(flow1, flow2):
+    flow1_d, flow2_d = json.loads(flow1), json.loads(flow2)
+    return all(flow1_d[field] == flow2_d[field] for field in fields)
+
+def flow_seen_in_flow_set(flow, flows):
+    return any(five_tuples_are_equal(flow, seen_flow) for seen_flow in flows)
+
+def hex_flow_counts_to_integers(flow):
+    to_modify = json.loads(flow)
+    to_modify['packet_count'] = int(to_modify['packet_count'], 16)
+    to_modify['flow_count'] = int(to_modify['flow_count'], 16)
+
+    return json.dumps(to_modify)
+
+def merge_flows(flow1, flow2):
+    flow1_d, flow2_d = json.loads(flow1), json.loads(flow2)
+    flow1_d['packet_count'] += flow2_d['packet_count']
+    flow1_d['flow_count'] += flow2_d['flow_count']
+
+    return json.dumps(flow1_d)
+
+def merge_cpu_flows(cpu_flows):
+    merged_flows = []
+    for flow in cpu_flows:
+        matching_flows = list(filter(lambda f: five_tuples_are_equal(f, flow), cpu_flows))
+        matching_flows.remove(flow) # The current flow should not match itself
+        merged_flow = f'{flow}'
+        for matching_flow in matching_flows:
+            merged_flow = merge_flows(merged_flow, matching_flow)
+        merged_flows.append(merged_flow)
+
+    return set(merged_flows) # Remove duplicate merged flows
+
+
+def singledecode(flow_info, hosts):
+    all_cpus_identified_flows = set()
+    for cpu in range(0, CPUS):
```

Comment:
can you just do `for cpu in flow_info.keys()`?

Reply:
yep